src/dymaptic.GeoBlazor.Core/Model/OAuthAuthentication.cs

```diff
@@ -0,0 +1,100 @@
+using dymaptic.GeoBlazor.Core.Exceptions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.JSInterop;
+
+
+namespace dymaptic.GeoBlazor.Core.Model;
+
+/// <summary>
+///     A simple class for handling OAuth authentication.
+/// </summary>
+public class OAuthAuthentication : LogicComponent
+{
+    /// <summary>
+    ///     Default Constructor
+    /// </summary>
+    /// <param name="jsRuntime">
+    ///     Injected JavaScript Runtime reference
+    /// </param>
+    /// <param name="config">
+    ///     Injected configuration object
+    /// </param>
+    public OAuthAuthentication(IJSRuntime jsRuntime, IConfiguration config) : base(jsRuntime, config)
+    {
+        _config = config;
+    }
+
+    /// <inheritdoc />
+    protected override string ComponentName => nameof(OAuthAuthentication);
+    private string? AppId => _config["ArcGISAppId"];
+    private string? PortalUrl => _config["ArcGISPortalUrl"];
+    
+
+    /// <summary>
+    ///     Tests to see if the user is logged in. True if yes, false if otherwise.
+    /// </summary>
+    /// <returns>
+    ///     Returns a boolean value indicating whether or not the user is logged in.
+    /// </returns>
+    public async Task<bool> IsLoggedIn()
+    {
+        return await InvokeAsync<bool>("isLoggedIn");
+    }
+
+    /// <summary>
+    ///     Logs the user in.
+    /// </summary>
+    public async Task Login()
+    {
+        await InvokeVoidAsync("doLogin");
+    }
+
+    /// <summary>
+    ///     Ensures that the user is logged in. If not, it will attempt to log them in.
+    /// </summary>
+    public async Task EnsureLoggedIn()
+    {
+        if (!await IsLoggedIn())
+        {
+            await Login();
+        }
+    }
+
+    /// <summary>
+    ///     Returns the current OAuth Token for use in requests.
+    /// </summary>
+    /// <returns>
+    ///     A string containing the current OAuth Token.
+    /// </returns>
+    public async Task<string> GetCurrentToken()
+    {
+        return await InvokeAsync<string>("getToken");
```

Comment:
I was kind of hoping that we could centralize the "token" management such that the dev could just call "getCurrentToken" and get back the api key if that's the current token, or the oauth token if that's the current token. That way there is a central point to extract the current token if you need to pass it to a custom rest call.

Reply:
This was good feedback and pushed me to do a bit more refactoring. I added an `ApiKeyAuthentication` class, moved all the APIKey logic out of `MapView` and into this `LogicComponent`, and then created a new `AuthenticationManager` class that checks for both api keys and app ids, initializes the correct logic component, and exposes `GetCurrentToken` as you suggested.